### DISCUSSION TITLE
Keep-alive/Keep trying and downloading future Twitch streams

### DISCUSSION THREAD

**Discussion by Infinitay** [streamlink/streamlink#6224] [Q&A] [answered]
### Description

Is there a flag that will make streamlink to not exit after a stream is over and to keep running the process for continuing and future streams? I would like to avoid creating a bat or using a scheduler if possible.

**Comment by bastimeyer** [comment, marked as answer]
No, there is not. There's [`--retry-streams`](https://streamlink.github.io/cli.html#cmdoption-retry-streams) for waiting until a stream becomes available, but that's it.

Streamlink's main purpose is watching live streams, not recording, so when a stream ends, it'll exit because its job is done.

You'll have to write a wrapper shell script and re-launch the process. Be aware of #4411 / #4783 though when waiting for Twitch streams to go live in order to record them.

**Reply by Infinitay** [reply]
Thanks for the answer. It makes total sense with the current purpose and functionality. I'll look into a scheduler or worst case creating a batch file for it.

> Be aware of https://github.com/streamlink/streamlink/issues/4411 / https://github.com/streamlink/streamlink/issues/4783 though when waiting for Twitch streams to go live in order to record them.

~~I came across this before and never knew what the case was and manually modified it myself. I'm curious if StreamLink can mitigate this issue by querying for the data until we get a non-null response and renaming the file once the stream is over and we've finished writing to the file.~~

I just read the end of https://github.com/streamlink/streamlink/issues/4411#issuecomment-1081965952 and you gave your input on this exact thing. Thanks again.
